### DISCUSSION TITLE
Way to reuse requests between scenarios?

### DISCUSSION THREAD

**Discussion by mnylensc** [artilleryio/artillery#3394] [Artillery]
Hey!

I'm working on a load test script and there's some parts that are duplicated in all scenarios within the same test script. This is like three different HTTP requests that all scenarios need to make.

Is there any way to remove the duplication from the test script? I was hoping there'd be a way to define a flow at top level and then reference it in the scenario, but looking at the docs, there doesn't seem to be support for something like that.

**Comment by hassy** [comment]
hi @mnylensc 👋 Are those 3 requests always exactly the same and are always made in the same order? One way of doing it would be to use YAML anchors and references - similar to the example here: https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/

**Reply by mnylensc** [reply]
Hmm, yeah. This would mostly work I think. There's one minor difference between the scenarios where one of them sets an existing session cookie for the requests and the other one doesn't. I guess I could handle this with a function somehow.

**Comment by jamesmyers-blc** [comment]
Here's an example because I haven't seen any examples of this within Artillery.io discussions etc

```
# note that the get is not prefixed with - as gets parsed as an object
.first: &first
  get: 
    url: /pathone

.second: &second
  get:
    url: /pathtwo

.mergedFlows: &mergedFlows
- *first
- *second

config:
   #some config goes here

scenarios:
  - name: "Merge yaml job templates"
    flow: *mergedFlows
```
edit:
Although to be honest whilst this executes the reporting seems to treat the initial declarations as it's own phase as well as a couple of other limitations.

**Reply by hassy** [reply]
thanks for sharing @jamesmyers-blc!
